### ISSUE TITLE
_create_ctr_cipher() raises an error when using an init vector with leading zero

### ISSUE THREAD

**Issue by jeansastre** [Legrandin/pycryptodome#491]
To reproduce:

```
from Crypto.Cipher import AES
from Crypto.Util import Counter
import binascii

key = bytearray.fromhex("1234567890abcdef1234567890abcdef")

iv_fails = bytes(range(16))
# iv_work = bytes(range(16, 32))

ctr = Counter.new(AES.block_size, initial_value=int(binascii.hexlify(iv_fails), 16))
aes = AES.new(key, AES.MODE_CTR, counter=ctr)

```

**Comment by texadactyl** [comment]
It seems that for each additional 0x00 byte in the IV, from left to right, the counter block is diagnosed to be smaller by 1 until you reach the 2nd to the last byte.  For the last 2 positions, the length is always 2.
```
from binascii import hexlify
from Crypto.Cipher import AES
from Crypto.Util import Counter

def dumper(arg_label, arg_data):
    print(arg_label, '({}): {}: {}'.format(len(arg_data), type(arg_data), arg_data))

def try_it(arg_iv):
    dumper('\nIV', arg_iv)
    key = bytearray.fromhex("1234567890abcdef1234567890abcdef")
    ctr = Counter.new(AES.block_size, initial_value=int(hexlify(arg_iv), 16))
    try:
        aes = AES.new(key, AES.MODE_CTR, counter=ctr)
        print('No exceptions reported')
    except ValueError as ex:
        print('OUCH!', ex)


try_it(bytes(b'\xff\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f'))
try_it(bytes(b'\x00\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f'))
try_it(bytes(b'\x00\x00\x12\x13\x14\x15\x16\x17\x18\x19\x1a\x1b\x1c\x1d\x1e\x1f'))
try_it(bytes(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x1a\x1b\x1c\x1d\x1e\x1f'))
try_it(bytes(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x03\x02\x01'))
try_it(bytes(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x02\x01'))
try_it(bytes(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01'))
try_it(bytes(b'\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00'))

```

`

**Comment by Varbin** [comment]
The counter wants its first argument in bits, not bytes:  `Counter.new(AES.block_size*8, initial_value=...)` (https://pycryptodome.readthedocs.io/en/latest/src/util/util.html?highlight=Counter.new#Crypto.Util.Counter.new).

For some reasons, PyCryptodome "extends" the bitsize, if the bitlength of `initial_value` is larger than `nbits`. Personally, I would expect PyCryptodome to either throw an error on Counter initialization or shorten the counter to the given bitlength.

**Comment by Legrandin** [comment]
> The counter wants its first argument in bits, not bytes: `Counter.new(AES.block_size*8, initial_value=...) `

Indeed.

> I would expect PyCryptodome to either throw an error on Counter initialization or shorten the counter to the given bitlength.

The error message is unhelpful. It should clearly say that the Counter is too small.

**Comment by texadactyl** [comment]
I overlooked that in the API document, myself.  With that fix, no more exceptions.  Agree about a diagnostic.

Thanks.  I'll be watching out for more bit vs byte cases.

**Comment by Legrandin** [comment]
There is now a fix with https://github.com/Legrandin/pycryptodome/commit/21ce2a72d36ceb73f1660cf94e7b56826b4b967d: `Counter.new` will fail with an appropriate error message if `initial_value` doesn't fit into `nbits` bits.

**Comment by Legrandin** [comment]
This fix landed in version 3.10.1
